src/test/fio/fio_ceph_messenger.cc

```diff
@@ -80,7 +80,8 @@ struct ceph_msgr_reply_io {
 
 static void *str_to_ptr(const std::string &str)
 {
-  return (void *)strtoul(str.c_str(), NULL, 16);
+  // if str is invalid pointer, may be a logic error, and allow crash
+  return (void *)strtoull(str.c_str(), NULL, 16);
```

Comment:
`strtoull`-based version would never crash.  Adding a comment was suggested in connection with switching to `ceph::parse` which may throw an exception:

```
// str is assumed to be a valid ptr string
return reinterpret_cast<void*>(ceph::parse<uintptr_t>(str, 16).value());
```

Reply:
Looks good but please squash everything into the first commit (the one with the `Signed-off-by`).